sunpy/data/_sample.py

```diff
@@ -166,3 +173,31 @@ def download_sample_data(overwrite=False):
             _handle_final_errors(results)
 
     return results + already_downloaded
+
+
+def _get_sample_files(filename_list, no_download=False, force_download=False):
+    sampledata_dir = _get_sampledata_dir()
+
+    fullpath = [sampledata_dir/fn for fn in filename_list]
+
+    if no_download:
+        fullpath = [fp if fp.exists() else None for fp in fullpath]
```

Comment:
It seems a bit un-intuitive that this returns a list that can include `None` entries - can you include a quick description of this behaviour in a short docstring to avoid future developer confusion?

Reply:
Added in two places